src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformSearchService.kt

```diff
@@ -0,0 +1,174 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexmanagement.transform
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.elasticapi.retry
+import com.amazon.opendistroforelasticsearch.indexmanagement.elasticapi.suspendUntil
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.exceptions.TransformSearchServiceException
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.model.Transform
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.model.TransformSearchResult
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.model.TransformStats
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.settings.TransformSettings.Companion.TRANSFORM_JOB_SEARCH_BACKOFF_COUNT
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.settings.TransformSettings.Companion.TRANSFORM_JOB_SEARCH_BACKOFF_MILLIS
+import com.amazon.opendistroforelasticsearch.indexmanagement.util.IndexUtils.Companion.ODFE_MAGIC_NULL
+import com.amazon.opendistroforelasticsearch.indexmanagement.util.IndexUtils.Companion.hashToFixedSize
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.DocWriteRequest
+import org.elasticsearch.action.bulk.BackoffPolicy
+import org.elasticsearch.action.index.IndexRequest
+import org.elasticsearch.action.search.SearchRequest
+import org.elasticsearch.action.search.SearchResponse
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.search.aggregations.Aggregation
+import org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregation
+import org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregationBuilder
+import org.elasticsearch.search.aggregations.bucket.composite.CompositeValuesSourceBuilder
+import org.elasticsearch.search.aggregations.metrics.InternalAvg
+import org.elasticsearch.search.aggregations.metrics.InternalMax
+import org.elasticsearch.search.aggregations.metrics.InternalMin
+import org.elasticsearch.search.aggregations.metrics.InternalSum
+import org.elasticsearch.search.aggregations.metrics.InternalValueCount
+import org.elasticsearch.search.aggregations.metrics.NumericMetricsAggregation
+import org.elasticsearch.search.aggregations.metrics.Percentiles
+import org.elasticsearch.search.aggregations.metrics.ScriptedMetric
+import org.elasticsearch.search.builder.SearchSourceBuilder
+import kotlin.math.max
+import kotlin.math.pow
+import org.elasticsearch.ExceptionsHelper
+import org.elasticsearch.index.query.QueryBuilder
+import org.elasticsearch.transport.RemoteTransportException
+
+class TransformSearchService(
```

Comment:
Any unit tests for this class? Or being added later as integ tests once working system?

Reply:
Thinking about adding integ tests as once working system